File: Praktikum/Praktikum 1/Kompleks.cpp
#include <iostream>
#include "Kompleks.h"
using namespace std;

int Kompleks::n_kompleks = 0;

Kompleks::Kompleks(){
	this->real = 0;
	this->imaginer = 0;
	n_kompleks++;
}

Kompleks::Kompleks(int real, int imaginer){
	this->real = real;
	this->imaginer = imaginer;
	n_kompleks++;
}

int Kompleks::GetReal() const{
	return this->real;
}

int Kompleks::GetImaginer() const{
	return this->imaginer;
}

void Kompleks::SetReal(int a){
	this->real = a;
}

void Kompleks::SetImaginer(int a){
	this->imaginer = a;
}

Kompleks operator+(const Kompleks& a, const Kompleks&b){
	Kompleks c;
	int A = a.GetReal();
	int B = a.GetImaginer();
	int C = b.GetReal();
	int D = b.GetImaginer();
	c.SetReal(A+C);
	c.SetImaginer(B+D);
	return c;
}

Kompleks operator-(const Kompleks& a, const Kompleks&b){
	Kompleks c;
	c.SetReal(A-C);
	c.SetImaginer(B-D);
	return c;
}

Kompleks operator*(const Kompleks& a, const Kompleks&b){
	Kompleks c;
	int A = a.GetReal();
	int B = a.GetImaginer();
	int C = b.GetReal();
	int D = b.GetImaginer();
	c.SetReal(A*C-(B*D));
	c.SetImaginer(B*C+(A*D));
	return c;
}

Kompleks operator*(const Kompleks& a, const int integer){
	Kompleks c;
	int A = a.GetReal();
	int B = a.GetImaginer();
	c.SetReal(A*integer);
	c.SetImaginer(B*integer);
	return c;
}

Kompleks operator*(const int integer, const Kompleks& a){
	Kompleks c;
	int A = a.GetReal();
	int B = a.GetImaginer();
	c.SetReal(A*integer);
	c.SetImaginer(B*integer);
	return c;
}

int Kompleks::CountKompleksInstance(){
	return n_kompleks;
};

void Kompleks::Print(){
	if (this->imaginer < 0){
		cout << this->real << this->imaginer << "i" << endl;
	} else {
		cout << this->real << "+" << this->imaginer << "i" << endl;
	} 
}

File: Responsi/Responsi 4/No 1/Kompleks.java
public class Kompleks {
    private static int n_kompleks = 0;
    private int real;
    private int imaginer;

    public Kompleks(int real, int imaginer) {
        this.real = real;
        this.imaginer = imaginer;
        n_kompleks++;
    }

    public Kompleks() {
        this.real = 0;
        this.imaginer = 0;
        n_kompleks++;
    }

    public int getReal() {
        return this.real;
    }

    public int getImaginer() {
        return imaginer;
    }

    public void setReal(int real) {
        this.real = real;
    }

    public void setImaginer(int imaginer) {
        this.imaginer = imaginer;
    }

    public Kompleks plus(Kompleks b){
        Kompleks c = new Kompleks();
        c.setReal(this.real + b.getReal());
        c.setImaginer(this.imaginer + b.getImaginer());
        return c;
    }

    public Kompleks minus(Kompleks b){
        Kompleks c = new Kompleks();
        c.setReal(this.real - b.getReal());
        c.setImaginer(this.imaginer - b.getImaginer());
        return c;
    }

    public Kompleks multiply(Kompleks b){
        Kompleks c = new Kompleks();
        c.setReal(this.real * b.getReal() - this.imaginer * b.getImaginer());
        c.setImaginer(this.real * b.getImaginer() + this.imaginer * b.getReal());
        return c;
    }

    public Kompleks multiply(int c){
        Kompleks d = new Kompleks();
        d.setReal(this.real * c);
        d.setImaginer(this.imaginer * c);
        return d;
    }

    public Kompleks multiply(int c, Kompleks b){
        return b.multiply(c);
    }

    public static int countKompleksInstance() {
        return n_kompleks;
    }

    public void print() {
        if (real != 0) {
            System.out.print(real);
        }
        if (imaginer > 0 && real != 0) {
            System.out.print("+");
        }
        if (imaginer != 0) {
            System.out.print(imaginer + "i");
        }
        if (real == 0 && imaginer == 0) {
            System.out.print(0);
        }
        System.out.println();
    }
}